pkg/controller/certrotation/certrotation_controller.go

```diff
@@ -0,0 +1,464 @@
+	mcsTLSKeyRefresh = mcsCARefresh
+
+	// This is used to key in to the user-data secret
+	userDataKey = "userData"
+
+	// ign* are for the user-data ignition fields
+	ignFieldIgnition = "ignition"
+	ignFieldSource   = "source"
+)
+
+type CertRotationController struct {
+	kubeClient   kubernetes.Interface
+	configClient configclientset.Interface
+
+	mcoConfigMapInfomer coreinformersv1.ConfigMapInformer
+	maoSecretInformer   coreinformersv1.SecretInformer
+
+	mcoSecretLister corelisterv1.SecretLister
+	maoSecretLister corelisterv1.SecretLister
+
+	certRotators []factory.Controller
+
+	recorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+}
+
+// New returns a new cert rotation controller.
+func New(
+	kubeClient kubernetes.Interface,
+	configClient configclientset.Interface,
+	machineClient machineclientset.Interface,
+	maoSecretInformer coreinformersv1.SecretInformer,
+	mcoSecretInformer coreinformersv1.SecretInformer,
+	mcoConfigMapInfomer coreinformersv1.ConfigMapInformer,
+) (*CertRotationController, error) {
+
+	recorder := events.NewLoggingEventRecorder(componentName)
+
+	c := &CertRotationController{
+		kubeClient:          kubeClient,
+		configClient:        configClient,
+		recorder:            recorder,
+		maoSecretInformer:   maoSecretInformer,
+		mcoConfigMapInfomer: mcoConfigMapInfomer,
+		mcoSecretLister:     mcoSecretInformer.Lister(),
+		maoSecretLister:     maoSecretInformer.Lister(),
+		cachesToSync: []cache.InformerSynced{
+			maoSecretInformer.Informer().HasSynced,
+			mcoSecretInformer.Informer().HasSynced,
+			mcoConfigMapInfomer.Informer().HasSynced,
+		},
+	}
+
+	// This is required for the machine-config-server-tls secret rotation
+	cfg, err := configClient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cluster infrastructure resource: %w", err)
+	}
+
+	serverIPs := getServerIPsFromInfra(cfg)
+
+	if cfg.Status.APIServerInternalURL == "" {
+		return nil, fmt.Errorf("no APIServerInternalURL found in cluster infrastructure resource")
+	}
+	apiserverIntURL, err := url.Parse(cfg.Status.APIServerInternalURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %w", apiserverIntURL, err)
+	}
+
+	// The cert controller will begin creating "machine-config-server-ca" configmap & secret in the MCO namespace.
+	// The *-user-data secrets will be updated based on these configmap/secrets.
+	// For the *-user-data-managed secrets, the operator will begin to use "machine-config-server-ca" configmap
+	// instead of root-CA bundle configmap
+
+	machineConfigServerCertRotator := certrotation.NewCertRotationController(
+		"MachineConfigServerCertRotator",
+		certrotation.RotatedSigningCASecret{
+			Namespace: ctrlcommon.MCONamespace,
+			Name:      ctrlcommon.MachineConfigServerCAName,
+			AdditionalAnnotations: certrotation.AdditionalAnnotations{
+				JiraComponent: "machine-config-operator",
```

Comment:
```suggestion
				JiraComponent: "Machine Config Operator",
```
This needs to match Jira component

Reply:
Fixed, thanks for catching that!

Reply:
You'd also need to update installer repo - https://github.com/openshift/installer/pull/9309